app/frontend/src/features/communities/CommunityPage/DiscussionCard.tsx

```diff
@@ -0,0 +1,120 @@
+import { makeStyles, Typography } from "@material-ui/core";
+import { Skeleton } from "@material-ui/lab";
+import classNames from "classnames";
+import { Error as GrpcError } from "grpc-web";
+import React, { useMemo } from "react";
+import { useInfiniteQuery } from "react-query";
+
+import { Card, CardActionArea, CardContent } from "../../../components/Card";
+import CircularProgress from "../../../components/CircularProgress";
+import { Discussion } from "../../../pb/discussions_pb";
+import { GetThreadRes } from "../../../pb/threads_pb";
+import { threadKey } from "../../../queryKeys";
+import { service } from "../../../service";
+import { timestamp2Date } from "../../../utils/date";
+import { firstName } from "../../../utils/names";
+import stripMarkdown from "../../../utils/stripMarkdown";
+import { timeAgo } from "../../../utils/timeAgo";
+import useUsers, { useUser } from "../../userQueries/useUsers";
+
+const useStyles = makeStyles((theme) => ({
+  root: {
+    width: "100%",
+  },
+  userLoading: { display: "inline-block", width: 80 },
+  title: {},
+  surtitle: { marginBottom: theme.spacing(0.5) },
+  replies: {
+    "&:first-child": { marginTop: theme.spacing(1) },
+  },
+}));
+
+export default function DiscussionCard({
+  discussion,
+  className,
+}: {
+  discussion: Discussion.AsObject;
+  className?: string;
+}) {
+  const classes = useStyles();
+  const { data: creator } = useUser(discussion.creatorUserId);
+  //although we are only using 1 page of query, still use infinite as that
```

Comment:
Does that mean this query is being/going to be reused elsewhere that need to be multiple pages? Otherwise what's the advantage of this over a normal `useQuery`? Not sure I follow the "this is how its stored" comment.

Reply:
Yes, I thought something will break if `["thread", threadId]` is used for both an infinite query and a normal query (no `pages` in a normal query).